PR: https://github.com/odoo/odoo/pull/236080

From: b28fa9768765d118519891b08e560d7e0a409c2b
From: alan-odoo
Date: 2026-09-02 08:30:17

Breaking data model changes scores: del:2 + add:0, change matches:
-    invoiced = fields.Float(
-    invoiced_target = fields.Float(

Total Changes: 45

[REM] sale{_project}: remove unnecessary fields and widget

The CrmTeam.invoiced and CrmTeam.invoiced_target fields are removed as they
have no utility. They were used by the progress bar in the sales team kanban
which has been removed in https://github.com/odoo/odoo/pull/203918. For the
same reason, the "sales_team_progressbar" widget is not being used and has
therefore also been removed.

Enterprise PR: https://github.com/odoo/enterprise/pull/99697
Upgrade PR: https://github.com/odoo/upgrade/pull/8989

Task-5258611

Part-of: odoo/odoo#236080
Related: odoo/enterprise#99697
Related: odoo/upgrade#8989
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/sale/models/crm_team.py
+++ b/addons/sale/models/crm_team.py
@@ -2,55 +2,13 @@
 
 from odoo import api, fields, models
 from odoo.exceptions import UserError
-from odoo.tools import SQL
 
 
 class CrmTeam(models.Model):
     _inherit = "crm.team"
 
-    invoiced = fields.Float(
-        compute="_compute_invoiced",
-        string="Invoiced This Month",
-        readonly=True,
-        help="Invoice revenue for the current month. This is the amount the sales "
-        "channel has invoiced this month. It is used to compute the progression ratio "
-        "of the current and target revenue on the kanban view.",
-    )
-    invoiced_target = fields.Float(
-        string="Invoicing Target",
-        help="Revenue Target for the current month (untaxed total of paid invoices).",
-    )
     sale_order_count = fields.Integer(compute="_compute_sale_order_count", string="# Sale Orders")
 
-    def _compute_invoiced(self):
-        if self.ids:
-            today = fields.Date.context_today(self)
-            data_map = dict(
-                self.env.execute_query(
-                    SQL(
-                        """
-                            SELECT move.team_id AS team_id,
-                                   SUM(move.amount_untaxed_signed) AS amount_untaxed_signed
-                              FROM account_move move
-                             WHERE move.move_type IN ('out_invoice', 'out_refund', 'out_receipt')
-                               AND move.payment_state IN ('in_payment', 'paid', 'reversed')
-                               AND move.state = 'posted'
-                               AND move.team_id IN %s
-                               AND move.date BETWEEN %s AND %s
-                          GROUP BY move.team_id
-                        """,
-                        tuple(self.ids),
-                        fields.Date.to_string(today.replace(day=1)),
-                        fields.Date.to_string(today),
-                    )
-                )
-            )
-        else:
-            data_map = {}
-
-        for team in self:
-            team.invoiced = data_map.get(team._origin.id, 0.0)
-
     def _compute_sale_order_count(self):
         sale_order_data = self.env["sale.order"]._read_group(
             [("team_id", "in", self.ids), ("state", "!=", "cancel")], ["team_id"], ["__count"]
@@ -74,9 +32,6 @@ class CrmTeam(models.Model):
             )
         return super().action_primary_channel_button()
 
-    def update_invoiced_target(self, value):
-        return self.write({"invoiced_target": round(float(value or 0))})
-
     @api.ondelete(at_uninstall=False)
     def _unlink_except_used_for_sales(self):
         """If more than 5 active SOs, we consider this team to be actively used.
